sympy/core/function.py

```diff
@@ -1005,10 +1005,15 @@ def __new__(cls, expr, *variables, **assumptions):
             variables = expr.free_symbols
             if len(variables) != 1:
                 from sympy.utilities.misc import filldedent
-                raise ValueError(filldedent('''
-                    Since there is more than one variable in the
-                    expression, the variable(s) of differentiation
-                    must be supplied to differentiate %s''' % expr))
+                if len(variables) == 0:
```

Comment:
@mafiya69 This is not supported. The argument to ``diff`` has to be a ``symbol``. ``t.diff(t)`` is 1 when ``t`` is a ``symbol``. In that case ``t.free_symbols`` is already ``set([t])``. +1 to @smichr comment.